source/postprocess/binary_data.cc

```diff
@@ -43,66 +48,87 @@ namespace aspect {
 //                               "option to support checkpoint/restart, but deal.II "
 //                               "did not detect its presence when you called 'cmake'."));
 //#endif
-       }
-
-       template <int dim>
-       std::pair<std::string, std::string> BinaryData<dim>::execute(TableHandler &statistics) {
-          std::string fileName = this->get_output_directory() + "/" + filename_prefix + Utilities::int_to_string(this->get_timestep_number(), 5) + ".mesh";
-          parallel::distributed::SolutionTransfer<dim, LinearAlgebra::BlockVector> sol_trans(this->get_dof_handler());
-          sol_trans.prepare_serialization (this->get_solution());
-          this->get_triangulation().save(fileName.c_str());
-
-/*           std::ofstream output (this->get_output_directory() + "solution-" + Utilities::int_to_string(this->get_timestep_number(), 5) + ".txt");
-           dealii::BlockVector<double> solution(this->get_solution());
-           solution.block_write(output);
-*/
-           return std::make_pair("Writing binary output to: ", fileName);
-       }
-
-        template <int dim>
-        void
-        BinaryData<dim>::declare_parameters(ParameterHandler &prm)
+    }
+
+    template< int dim>
+    void BinaryData<dim>::update_time()
+    {
+      attributes.time = this->get_time();
+      attributes.time_step = this->get_timestep();
+      attributes.old_time_step = this->get_old_timestep();
+      attributes.timestep_number = this->get_timestep_number();
+    }
+
+    template <int dim>
+    std::pair<std::string, std::string> BinaryData<dim>::execute(TableHandler &statistics)
+    {
+      if (my_id == 0)
         {
-            prm.enter_subsection("Postprocess");
-            {
-                prm.enter_subsection("Binary data");
-                {
-                    prm.declare_entry("Binary data file name", "solution-",
-                    Patterns::FileName(),
-                    "A file name prefix that will be appended to each file that is generated by the "
-                    "binary output postprocessor.");
-                }
-                prm.leave_subsection();
-            }
-            prm.leave_subsection();
+          this->update_time();
+
+          std::ofstream ofs(this->get_output_directory() + "/" + filename_prefix +
+                            Utilities::int_to_string(this->get_timestep_number(), 5) + ".time");
+          boost::archive::binary_oarchive oa(ofs);
+          oa << attributes;
```

Comment:
This is writing out the field data in binary format and when deserializing this data: the variables time, time step, and time step number will need to be set appropriately.  @tjhei ,where do you recommend adding this piece of code?